src/vos/vos_tree.c

```diff
@@ -749,6 +748,18 @@ svt_rec_update(struct btr_instance *tins, struct btr_record *rec,
 	return svt_rec_alloc_common(tins, rec, skey, rbund);
 }
 
+static int
+svt_rec_corrupt(struct btr_instance *tins, struct btr_record *rec)
+{
+	struct vos_irec_df	*irec;
+
+	irec = vos_rec2irec(tins, rec);
+	D_DEBUG(DB_IO, "Setting record bio_addr flag to corrupted\n");
+	BIO_ADDR_SET_CORRUPTED(&irec->ir_ex_addr);
```

Comment:
You did a in-place update (within PMDK transaction) without making snapshot by umem_tx_add().